src/sentry/workflow_engine/migration_helpers/rule.py

```diff
@@ -110,3 +115,110 @@ def create_if_dcg(
+        return
+
+    conditions, filters = split_conditions_and_filters(data["conditions"])
+
+    update_when_dcg(
+        when_dcg=workflow.when_condition_group,
+        conditions=conditions,
+        action_match=data["action_match"],
+        user=user,
+    )
+
+    try:
+        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
+    except WorkflowDataConditionGroup.DoesNotExist:
+        logger.exception(
+            "WorkflowDataConditionGroup does not exist", extra={"workflow_id": workflow.id}
+        )
+        # if no IF DCG exists, create one
+        if_dcg = create_if_dcg(
+            workflow=workflow, filters=filters, filter_match=data.get("filter_match")
+        )
+        WorkflowDataConditionGroup.objects.create(workflow=workflow, condition_group=if_dcg)
+
+    update_if_dcg(if_dcg=if_dcg, filters=filters, filter_match=data.get("filter_match"), user=user)
+
+    delete_workflow_actions(if_dcg=if_dcg, actions=data["actions"], user=user)
+    create_workflow_actions(if_dcg=if_dcg, actions=data["actions"])  # action(s) must exist
+
+    updated_data = {
+        "name": data["name"],
+        "environment_id": data["environment"],
+        "enabled": True,
+        "owner_team_id": None,
+        "owner_user_id": None,
+        "config": {"frequency": data.get("frequency", 30)},
+    }
+    owner = data.get("owner")
+    if owner:
+        if owner and owner.is_user:
+            updated_data["owner_user_id"] = owner.id
+        if owner and owner.is_team:
+            updated_data["owner_team_id"] = owner.id
+    workflow.update(**updated_data)
+
+
+def update_when_dcg(
+    when_dcg: DataConditionGroup,
+    conditions: list[dict[str, Any]],
+    action_match: str,
+    user: RpcUser | None = None,
+):
+    delete_data_conditions(dcg=when_dcg, user=user)
+
+    if when_dcg.logic_type != action_match:
+        if action_match == "any":
+            action_match = DataConditionGroup.Type.ANY_SHORT_CIRCUIT.value
+
+        when_dcg.update(logic_type=action_match)
+
+    for condition in conditions:
+        translate_to_data_condition(dict(condition), dcg=when_dcg)
+
+    return when_dcg
+
+
+def update_if_dcg(
+    if_dcg: DataConditionGroup,
+    filters: list[dict[str, Any]],
+    filter_match: str | None = None,
+    user: RpcUser | None = None,
+):
+    delete_data_conditions(dcg=if_dcg, user=user)
+
+    if if_dcg.logic_type != filter_match:
+        if filter_match == "any" or filter_match is None:
+            filter_match = DataConditionGroup.Type.ANY_SHORT_CIRCUIT.value
+
+        if_dcg.update(logic_type=filter_match)
+
+    for filter in filters:
+        translate_to_data_condition(dict(filter), dcg=if_dcg)
+
+    return if_dcg
+
+
+def delete_data_conditions(dcg: DataConditionGroup, user: RpcUser | None = None):
+    for data_condition in DataCondition.objects.filter(condition_group=dcg):
+        RegionScheduledDeletion.schedule(instance=data_condition, days=0, actor=user)
```

Comment:
hmm i'm not sure we need to schedule this since this can't be user-cancelable, also we'll wrap the updating of the rule+workflow in a transaction so it can be rolled back if something fails